### PR TITLE
Table panel: Use link elements instead of div elements with on click events to aid with keyboard accessibility

### LINKED ISSUE #59038: A11y: Fix keyboard accessibility in `packages/grafana-ui/src/components/Table/JSONViewCell.tsx`
https://github.com/grafana/grafana/blob/b2b0be7b939561541ece19f6679fad457f5fcdd1/packages/grafana-ui/src/components/Table/JSONViewCell.tsx#L40

### LINKED ISSUE #59041: A11y: Fix keyboard accessibility in `packages/grafana-ui/src/components/Table/ImageCell.tsx`
https://github.com/grafana/grafana/blob/b2b0be7b939561541ece19f6679fad457f5fcdd1/packages/grafana-ui/src/components/Table/ImageCell.tsx#L23

### LINKED ISSUE #59042: A11y: Fix keyboard accessibility in `packages/grafana-ui/src/components/Table/FilterPopup.tsx`
https://github.com/grafana/grafana/blob/ff1149ac39f50ead19ed67f8d70ac2c20a9a686b/packages/grafana-ui/src/components/Table/FilterPopup.tsx#L49

### LINKED ISSUE #59043: A11y: Fix keyboard accessibility in `packages/grafana-ui/src/components/Table/DefaultCell.tsx`
https://github.com/grafana/grafana/blob/b2b0be7b939561541ece19f6679fad457f5fcdd1/packages/grafana-ui/src/components/Table/DefaultCell.tsx#L41

### PR DESCRIPTION
**Pull request by oscarkilhed**
**What is this feature?**
Changes some div elements with on click events into link elements to aid with keyboard navigation

**Which issue(s) does this PR fix?**:
Fixes #59043
Fixes #59042
Fixes #59041
Fixes #59038

### CODE REVIEW

**Review by mckn — APPROVED** [review]
LGTM!

**Comment by oscarkilhed** [comment]
So this got stuck for a while because the `DataLinkContextMenu` would nest the element inside an anchor tag, which in turn violates the HTML rule that an anchor tag cannot be a child of an anchor tag.

File: packages/grafana-ui/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -311,3 +311,18 @@ export const clearButtonStyles = (theme: GrafanaTheme2) => {
     padding: 0;
   `;
 };
+
+export const clearLinkButtonStyles = (theme: GrafanaTheme2) => {
+  return css`
+    background: transparent;
+    border: none;
+    padding: 0;
+    font-family: inherit;
```
Comment (baldm0mma) [inline packages/grafana-ui/src/components/Button/Button.tsx:320]: Are `color` and `font-family` not already inherited attrs by default?

File: packages/grafana-ui/src/components/Table/ImageCell.tsx [hunk 2]
```diff
@@ -12,18 +14,23 @@ export const ImageCell: FC<TableCellProps> = (props) => {
...
           {(api) => {
-            return (
-              <div onClick={api.openMenu} className={cx(tableStyles.imageCellLink, api.targetClassName)}>
-                <img src={displayValue.text} className={tableStyles.imageCell} alt="" />
-              </div>
-            );
+            if (api.openMenu) {
+              return (
+                <Button className={cx(clearButtonStyle)} onClick={api.openMenu}>
+                  <img src={displayValue.text} className={tableStyles.imageCell} alt="" />
```
Comment (baldm0mma) [inline packages/grafana-ui/src/components/Table/ImageCell.tsx:28]: Do we need a more descriptive text here for the `alt` prop for better accessibility?

File: packages/grafana-ui/src/components/Button/Button.tsx (hunk 1 above)
Comment (oscarkilhed) [reply packages/grafana-ui/src/components/Button/Button.tsx:320]: Yes. But they are set with emotion on the Button element. The final css class generated for the button is going to look a little weird, it's going to have two entries for both color and font-family. The first one being a specific color/font (set by the styling in the button element) then these lines that gets tacked on and overrides those entries making the button inherit the font and color from the parent element.

File: packages/grafana-ui/src/components/Table/ImageCell.tsx (hunk 2 above)
Comment (oscarkilhed) [reply packages/grafana-ui/src/components/Table/ImageCell.tsx]: I don't think we can set something useful for the alt text here as the image url is set from the value of the field. The only thing we could set it to would be the url, or maybe a substring of the url. Probably not something that would be very helpful to someone useing a screen reader.

**Pull request merged**